File: src/main/java/com/github/graqr/threshr/model/redsky/products/pdp/client/EligibilityRules.java
package com.github.graqr.threshr.model.redsky.products.pdp.client;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.serde.annotation.Serdeable;

@Serdeable
public record EligibilityRules(
        @JsonProperty("add_on")
        AddOn addOn,
        Grocery grocery,
        Hold hold,
        @JsonProperty("inventory_notification_to_guest_excluded")
        InventoryNotificationToGuestExcluded inventoryNotificationToGuestExcluded,
        @JsonProperty("scheduled_delivery")
        ScheduledDelivery scheduledDelivery) {
}

File: src/main/kotlin/com/graqr/threshr/model/redsky/product/EligibilityRules.kt
package com.graqr.threshr.model.redsky.product

import com.fasterxml.jackson.annotation.JsonProperty
import io.micronaut.serde.annotation.Serdeable

@Serdeable
data class EligibilityRules(
    @param:JsonProperty("add_on") val addOn: AddOn,
    val grocery: Grocery,
    val hold: Hold,
    @param:JsonProperty("inventory_notification_to_guest_excluded") val inventoryNotificationToGuestExcluded: InventoryNotificationToGuestExcluded,
    @param:JsonProperty("scheduled_delivery") val scheduledDelivery: ScheduledDelivery
) {}
